homeassistant/components/device_tracker/xfinity.py

```diff
@@ -0,0 +1,67 @@
+"""
+Support for Xfinity Gateways.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.xfinity/
+"""
+import logging
+
+from requests.exceptions import RequestException
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.device_tracker import (
+    DOMAIN, PLATFORM_SCHEMA, DeviceScanner)
+from homeassistant.const import CONF_HOST
+
+REQUIREMENTS = ['xfinity-gateway==0.0.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_HOST = '10.0.0.1'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string
+})
+
+
+def get_scanner(hass, config):
+    """Validate the configuration and return an Xfinity Gateway scanner."""
+    info = config[DOMAIN]
```

Comment:
since there's a default for `CONF_HOST`, we can do this as a one-liner:

`scanner = XfinityDeviceScanner(config[DOMAIN][CONF_HOST])`